ui/src/app/components/routing.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+import {withRoute} from 'app/routes';
+import {history} from 'app/services/history';
+
+export const Router = withRoute()(({route}) => {
+  if (!route) {
+    return <div>Not found</div>;
+  }
+  return route.render();
+});
+
+export class Redirect extends React.Component<{to: any}> {
+  componentDidMount() {
+    const {to} = this.props;
+    history.replace(to);
+  }
+
+  render() {
+    return null;
+  }
+}
+
+// Example of how to enforce login on particular pages
```

Comment:
This is partly pseudo-code since the 'authState' concept hasn't been written yet, but it shows the basic concept of how to write a 'guard' by conditionally rendering children.